gspread/worksheet.py

```diff
@@ -399,6 +400,7 @@ def get_all_records(
         :param str value_render_option: (optional) Determines how values should
             be rendered in the the output. See `ValueRenderOption`_ in
             the Sheets API.
+        :param list expected_headers: (optional) List of expected headers, they must be unique.
```

Comment:
Consider adding something like: `Dictionaries will be returned for all headers, even if not included in this list.`